### ISSUE TITLE
FDEV_SETUP_STREAM not working with C++

### ISSUE THREAD

**Issue by sprintersb** [avrdudes/avr-libc#898]
Macro `FDEV_SETUP_STREAM` won't work with C++

```c++
#include <stdio.h>

FILE stream = FDEV_SETUP_STREAM (NULL, NULL, 0);
```
Error is because C++ requires that all components are initialized and in the order they appear in the struct definition:

```none
error: designator order for field '__file::flags' does not match declaration order in 'FILE' {aka '__file'}
```

Fix is to initialize all elements of `struct __file` in their order.

**Comment by sprintersb** [comment]
For the time being, the following work-around should do irrespective of lang version:
```c++
#include <stdio.h>

#undef FDEV_SETUP_STREAM
#define FDEV_SETUP_STREAM(PU, GE, FL) \
    {                                 \
        0  /* buf */,                 \
        0  /* unget */,               \
        FL /* flags */,               \
        0  /* size */,                \
        0  /* len */,                 \
        PU /* put */,                 \
        GE /* get */,                 \
        0  /* udata */                \
    }
```

**Comment by sprintersb** [comment]
Fixed by bda8c47c94e9ceadec041f91a34c61e2e5ea8209.
